buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -613,7 +613,7 @@ class BeamModulePlugin implements Plugin<Project> {
     def google_clients_version = "2.0.0"
     def google_cloud_bigdataoss_version = "2.2.26"
     // [bomupgrader] TODO(#35868): currently pinned, should be determined by: com.google.cloud:google-cloud-spanner, consistent with: google_cloud_platform_libraries_bom
-    def google_cloud_spanner_version = "6.95.1"
+    def google_cloud_spanner_version = "6.104.0"
```

Comment:
we can adjust the comment above "currently pinned before gcp-bom moving beyond this version")